src/storage/src/hummock/store/version.rs

```diff
@@ -433,6 +550,7 @@ impl HummockVersionReader {
             if let Some(data) = get_from_batch(
```

Comment:
L546 filters imm whose min epoch <= `min_epoch`, which seems incorrect. For example, if we have a merged imm with `epochs = [1, 2, 3]` and the provided `min_epoch = 2`, we should keep this merged imm.